eclipse-cdt-cloud/cdt-cloud-blueprint
# Add initial version of example generator

#### What it does

Adds a new extension that can generate example projects into the current workspace.
As an initial starting point, it adds a plain CMake project example with launch.json, the example project of [`clangd-contexts`](https://github.com/eclipse-cdt-cloud/clangd-contexts), and traces for TraceCompass.cloud.

The example generation is triggered via a command. In the arguments of this command you can optionally already pre-specify the example type.

This extension is extensible in the sense that you can add more examples by adding a new example folder and extending the command asking you which example to generate.

Also
* Fix getting started page in browser app
* Fix browser app and add default workspace

Contributed on behalf of STMicroelectronics.

#### How to test
  * Build and run the CDT.cloud blueprint app
  * Run the command `Generate CDT.cloud Blueprint Example`
  * It'll ask you which example to generate
  * It'll generate the example into the workspace
  * It'll open the `README.md`
  * Test whether the instructions of the shown readme work as expected

  * Show the welcome page
  * Find the link in the text to create an example project (it should open the command and let you select the example type)
  * Alternatively, there is also a section on the right in the welcome page that directly creates the CMake example
